android/src/main/java/com/pkmnapps/nearby_connections/NearbyConnectionsPlugin.java

```diff
@@ -386,6 +399,15 @@ public void onPayloadReceived(@NonNull String endpointId, @NonNull Payload paylo
                     // This is deprecated and only available on Android 10 and below.
                     args.put("filePath", payload.asFile().asJavaFile().getAbsolutePath());
                 }
+            } else if (payload.getType() == Payload.Type.STREAM) {
+                try {
+                    InputStream inputStream = payload.asStream().asInputStream(); 
+                    byte[] bytes = new byte[inputStream.available()];
+                    inputStream.read(bytes);
+                    args.put("bytes", bytes);
```

Comment:
A stream would contain an endless amount on data, `inputStream.read(bytes)` this method will probably only read the bytes available until it needs to wait. I am not sure, because if it doesn't it would freeze the thread until the stream ends.

Reply:
Here's another link that might ne useful: https://developers.google.com/nearby/connections/android/exchange-data#stream